traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -1017,8 +1025,8 @@ func updateV31(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, dsV31 *
 		}
 	}
 
-	if newDSType.HasSSLKeys() && (oldHostName != newHostName || oldCDNName != newCDNName || oldRoutingName != *ds.RoutingName) {
-		return nil, http.StatusBadRequest, errors.New("delivery service CDN is immutable"), errors.New("delivery service has ssl keys that cannot be automatically changed")
+	if newDSType.HasSSLKeys() && exists && (oldHostName != newHostName || oldCDNName != newCDNName || oldRoutingName != *ds.RoutingName) {
```

Comment:
kind of confused why `newDSType.HasSSLKeys()` is not enough to determine if ssl keys exist or not for this ds...

Reply:
to me this looks like "if has key and keys exist and....". seems redundant.